cognite_toolkit/_cdf_tk/loaders/_resource_loaders/industrial_tool_loaders.py

```diff
@@ -157,7 +158,14 @@ def retrieve(self, ids: SequenceNotStr[str]) -> StreamlitList:
         return StreamlitList([Streamlit.from_file(file) for file in files])
 
     def update(self, items: StreamlitWriteList) -> StreamlitList:
-        files = items.as_file_list()
+        files = []
+        for item in items:
+            content = self._as_json_string(item.external_id, item.entrypoint)
+            to_update = item.as_file()
+            to_update.metadata[self._metadata_hash_key] = calculate_str_or_file_hash(content, shorten=True)  # type: ignore[index]
```

Comment:
Doesn't this overwrite the metadata_hash_key in each iteration? Thus only counting the last iteration.